modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/VoskEngine.java

```diff
@@ -0,0 +1,152 @@
+import org.osgi.service.component.annotations.Modified;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.File;
+import java.net.URI;
+import java.util.ArrayList;
+import java.util.Dictionary;
+import java.util.List;
+import java.util.Map;
+
+/** Vosk implementation of the Speech-to-text engine interface. */
+@Component(
+    immediate = true,
+    service = {
+        SpeechToTextEngine.class,
+        ManagedService.class
+    },
+    property = {
+        "service.description=Vosk implementation of the SpeechToTextEngine interface",
+        "service.pid=org.opencastproject.speechtotext.impl.engine.VoskEngine"
+    }
+)
+public class VoskEngine implements SpeechToTextEngine, ManagedService {
+
+  private static final Logger log = LoggerFactory.getLogger(VoskEngine.class);
+
+  /** Name of the engine. */
+  private static final String engineName = "Vosk";
+
+  /** Config key for setting the path to the vosk. */
+  private static final String VOSK_EXECUTABLE_PATH_CONFIG_KEY = "vosk.root.path";
+
+  /** Default path to vosk. */
+  public static final String VOSK_EXECUTABLE_DEFAULT_PATH = "vosk-cli";
+
+  /** Currently used path of the vosk installation. */
+  private String voskExecutable = VOSK_EXECUTABLE_DEFAULT_PATH;
+
+
+  @Override
+  public String getEngineName() {
+    return engineName;
+  }
+
+  @Activate
+  public void activate(ComponentContext cc) {
+    log.debug("Activating Vosk as viable speech-to-text engine...");
+  }
+
+  /**
+   * OSGI callback when the configuration is updated. This method is only here to prevent the
+   * configuration admin service from calling the service deactivate and activate methods
+   * for a config update. It does not have to do anything as the updates are handled by updated().
+   */
+  @Modified
+  public void modified(Map<String, Object> config) throws ConfigurationException {
+    log.debug("Modified vosk engine service");
+  }
+
+  @Override
+  public void updated(Dictionary properties) throws ConfigurationException {
+    if (properties == null) {
+      return;
+    }
+    log.debug("Start updating Vosk configuration.");
+    voskExecutable = StringUtils.defaultIfBlank(
+            (String) properties.get(VOSK_EXECUTABLE_PATH_CONFIG_KEY), VOSK_EXECUTABLE_DEFAULT_PATH);
+    log.debug("Set vosk path to {}", voskExecutable);
+
+    log.debug("Finished updating Vosk configuration");
+  }
+
+  /**
+   * {@inheritDoc}
+   *
+   * @see org.opencastproject.speechtotext.api.SpeechToTextEngine#generateSubtitlesFile(URI, File, String)
+   */
+  @Override
+  public File generateSubtitlesFile(URI mediaFile, File preparedOutputFile, String language)
+          throws SpeechToTextEngineException {
+
+    final List<String> command = new ArrayList<>();
+    command.add(voskExecutable);
+    command.add("-i");
+    command.add(mediaFile.toString());
+    command.add("-o");
+    command.add(preparedOutputFile.getAbsolutePath());
+    command.add("-l");
+    command.add(language);
```

Comment:
Seems simpler, though this does not really matter and I don't have any strong feelings about wither way:

```suggestion
    final List<String> command = Arrays.asList(
      voskExecutable,
      "-i", mediaFile.toString(),
      "-o", preparedOutputFile.getAbsolutePath(),
      "-l", language);
```